tests/scheduler_test.rs

```diff
@@ -0,0 +1,60 @@
+use std::boxed::Box;
+use std::cell::RefCell;
+use std::rc::Rc;
+use yew::scheduler::{scheduler, Runnable, Scheduler};
+
+#[allow(missing_debug_implementations)]
+struct ImmediateRunnable {
+    output: Rc<RefCell<Vec<i32>>>,
+    value: i32,
+}
+
+impl Runnable for ImmediateRunnable {
+    fn run(self: Box<Self>) {
+        let mut output = self.output.borrow_mut();
+        output.push(self.value);
+    }
+}
+
+#[allow(missing_debug_implementations)]
+struct IndirectionRunnable {
+    output: Rc<RefCell<Vec<i32>>>,
+    scheduler: Rc<Scheduler>,
+}
+
+impl Runnable for IndirectionRunnable {
+    fn run(self: Box<Self>) {
+        self.scheduler.put_and_try_run(Box::new(ImmediateRunnable {
```

Comment:
Since scheduler is a static, it doesn't need to be passed around. You could do `let scheduler = scheduler();` here